lib/matrix.js

```diff
@@ -52,11 +58,351 @@ var init = function(exports){
     var HEADERS = {
         "User-Agent": "matrix-js"
     };
+    
+    // Basic DAOs to abstract slightly from the line protocol and let the
+    // application customise events with domain-specific info
+    // (e.g. chat-specific semantics) if it so desires.
+    
+    /*
+     * Construct a Matrix Event object
+     * @param {Object} event The raw event to be wrapped in this DAO
+     */
+    function MatrixEvent(event) {
+        this.event = event;
+    };
+    exports.MatrixEvent = MatrixEvent;
+    
+    MatrixEvent.prototype = {
+        getId: function() {
+            return this.event.event_id;
+        },
+        getSender: function() {
+            return this.event.user_id;
+        },
+        getType: function() {
+            return this.event.type;
+        },
+        getRoomId: function() {
+            return this.event.room_id;
+        },
+        getTs: function() {
+            return this.event.ts;
+        },
+        getContent: function() {
+            return this.event.content;
+        },
+        isState: function() {
+            return this.event.state_key !== undefined;
+        },
+    };
+    
+    function MatrixInMemoryStore() {
+        this.rooms = {
+            // state: { },
+            // timeline: [ ],
+        };
+        
+        this.presence = {
+            // presence objects keyed by userId
+        };
+    };
+    exports.MatrixInMemoryStore = MatrixInMemoryStore;
+        
+    // XXX: this is currently quite procedural - we could possibly pass back
+    // models of Rooms, Users, Events, etc instead.
+    MatrixInMemoryStore.prototype = {
+                
+        /*
+         * Add an array of one or more state MatrixEvents into the store, overwriting
+         * any existing state with the same {room, type, stateKey} tuple.
+         */
+        setStateEvents: function(stateEvents) {
```

Comment:
You have the single versions of most everything else but not this. I'd just accept ``stateEvents`` being an ``object`` then assigning it to an ``array`` if it is. Alternatively, you could add a ``setStateEvent`` method and then just call through to this.